kubernetes-addons.tf

```diff
@@ -155,6 +155,16 @@ module "argocd" {
   depends_on = [module.aws_eks]
 }
 
+module "aws_node_termination_handler" {
+  count                                   = var.create_eks && var.aws_node_termination_handler_enable && var.enable_self_managed_nodegroups ? 1 : 0
+  source                                  = "./kubernetes-addons/aws-node-termination-handler"
+  eks_cluster_name                        = module.aws_eks.cluster_id
+  aws_node_termination_handler_helm_chart = var.aws_node_termination_handler_helm_chart
+  autoscaling_group_names                 = var.create_eks && var.enable_self_managed_nodegroups ? values({ for nodes in sort(keys(var.self_managed_node_groups)) : nodes => join(",", module.aws_eks_self_managed_node_groups[nodes].self_managed_asg_names) }) : []
```

Comment:
consider making this value an export of the self managed node group module itself? Might be a bit cleaner.

Reply:
This is borrowed from [here](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/0f009136e2828ac537417123a55024f6a7e197cc/kubernetes-addons.tf#L142). I think its done this way because we don't know the keys for the `self_managed_node_groups`. Will defer to @vara-bonthu, is there a way to simplify this?

Reply:
I agree that we can move this to module output variable. It looks like we have multiple usecases for this. You can raise a backlog for this. We will deal with in next PR.